code/modules/mining/fulton.dm

```diff
@@ -6,136 +6,179 @@ GLOBAL_LIST_EMPTY(total_extraction_beacons)
-			sleep(1 SECONDS)
-			balloon3 = mutable_appearance('icons/effects/fulton_balloon.dmi', "fulton_retract")
-			balloon3.pixel_y = 10
-			balloon3.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
-			holder_obj.cut_overlay(balloon)
-			holder_obj.add_overlay(balloon3)
-			sleep(0.4 SECONDS)
-			holder_obj.cut_overlay(balloon3)
-			A.set_anchored(FALSE) // An item has to be unanchored to be extracted in the first place.
-			A.set_density(initial(A.density))
-			animate(holder_obj, pixel_z = 0, time = 5)
-			sleep(0.5 SECONDS)
-			A.forceMove(holder_obj.loc)
-			qdel(holder_obj)
-			if(uses_left <= 0)
-				qdel(src)
 
+	balloon_alert_to_viewers("attaching...")
+	playsound(thing, 'sound/items/zip.ogg', vol = 50, vary = TRUE)
+	if(isliving(thing))
+		var/mob/living/creature = thing
+		if(creature.mind)
+			to_chat(thing, span_userdanger("You are being extracted! Stand still to proceed."))
+
+	if(!do_after(user, 5 SECONDS, target = thing))
+		return
+
+	balloon_alert_to_viewers("extracting!")
+	if(loc == user)
+		user.back?.atom_storage?.attempt_insert(src, user, force = STORAGE_SOFT_LOCKED)
+	uses_left--
+
+	if(uses_left <= 0)
+		user.transferItemToLoc(src, thing, TRUE)
+
+	var/mutable_appearance/balloon
+	var/mutable_appearance/balloon2
+	var/mutable_appearance/balloon3
+
+	if(isliving(thing))
+		var/mob/living/creature = thing
+		creature.Paralyze(32 SECONDS) // Keep them from moving during the duration of the extraction
+		if(creature.buckled)
+			creature.buckled.unbuckle_mob(creature, TRUE) // Unbuckle them to prevent anchoring problems
+	else
+		thing.set_anchored(TRUE)
+		thing.set_density(FALSE)
+
+	var/obj/effect/extraction_holder/holder_obj = new(get_turf(thing))
+	holder_obj.appearance = thing.appearance
+	thing.forceMove(holder_obj)
+	balloon2 = mutable_appearance('icons/effects/fulton_balloon.dmi', "fulton_expand")
+	balloon2.pixel_y = 10
+	balloon2.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+	holder_obj.add_overlay(balloon2)
+
+	sleep(0.4 SECONDS)
+
+	balloon = mutable_appearance('icons/effects/fulton_balloon.dmi', "fulton_balloon")
+	balloon.pixel_y = 10
+	balloon.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+	holder_obj.cut_overlay(balloon2)
+	holder_obj.add_overlay(balloon)
+	playsound(holder_obj.loc, 'sound/items/fultext_deploy.ogg', vol = 50, vary = TRUE, extrarange = -3)
+
+	animate(holder_obj, pixel_z = 10, time = 2 SECONDS)
+	animate(pixel_z = 15, time = 1 SECONDS)
+	animate(pixel_z = 10, time = 1 SECONDS)
+	animate(pixel_z = 15, time = 1 SECONDS)
+	animate(pixel_z = 10, time = 1 SECONDS)
+	sleep(6 SECONDS)
+
+	playsound(holder_obj.loc, 'sound/items/fultext_launch.ogg', vol = 50, vary = TRUE, extrarange = -3)
+	animate(holder_obj, pixel_z = 1000, time = 3 SECONDS)
+
+	if(ishuman(thing))
+		var/mob/living/carbon/human/creature = thing
+		creature.SetUnconscious(0)
+		creature.remove_status_effect(/datum/status_effect/drowsiness)
+		creature.SetSleeping(0)
```

Comment:
did someone remove it making you scream at some point? it used to make you actually *scream and then yell over the radio